src/vtp/script_libs/cast_ballot_lib.py

```diff
@@ -0,0 +1,334 @@
+                )
+            else:
+                prompt = "Please enter the number for your choice: "
+            pyinputplus.inputCustom(validate_multichoice, prompt=prompt, blank=True)
+        else:
+            # Then prompt for input
+            prompt = "Please enter in rank order the numbers of your choices separated by spaces: "
+            pyinputplus.inputCustom(validate_multichoice, prompt=prompt, blank=True)
+
+    def loop_over_contests(self, a_ballot):
+        """Will loop over the contests in a ballot and either ask the user
+        for a choice or if in demo mode will randomly choose one.
+        """
+        contests = Contests(a_ballot)
+        total_contests = contests.len()
+        count = 0
+        contest_uids = []
+        for contest in contests:
+            contest_uids.append(contest.get("uid"))
+            if self.parsed_args.demo_mode:
+                self.make_random_selection(a_ballot, contest)
+            else:
+                # Display the tally type and choices and allow the user to manually
+                # enter something.  Might as well validate legal selections (in
+                # this demo) as that is the long-term VTP vision.
+                count += 1
+                self.get_user_selection(a_ballot, contest, count, total_contests)
+        if not self.parsed_args.demo_mode:
+            # UX wise replicate the self adjudication experince.  This is
+            # basically another endless loop until done
+            while True:
+                # Print the selections
+                for contest in contests:
+                    print(
+                        f"Contest {contest.get('uid')} - {contest.get('name')}: "
+                        f"{contest.get('selection')}"
+                    )
+                prompt = (
+                    "Is this correct?  "
+                    "Enter yes to accept the ballot, no to reject the ballot: "
+                )
+                if "yes" == pyinputplus.inputYesNo(prompt):
+                    break
+                prompt = (
+                    "Enter a contest uid to redo that contest, "
+                    "enter nothing (leave blank and hit enter) to start completely over: "
+                )
+                response = pyinputplus.inputChoice(contest_uids + [""], prompt)
+                if response == "":
+                    count = 0
+                    for contest in contests:
+                        count += 1
+                        self.get_user_selection(
+                            a_ballot, contest, count, total_contests
+                        )
+                else:
+                    for contest in contests:
+                        if contest.get("uid") == response:
+                            self.get_user_selection(a_ballot, contest, 1, 1)
+                            break
+        # For a convenient side effect, return the contests
+        return contests
+
```

Comment:
Move argument parsing (everything below this point) back to the script file (`accept_ballot.py` (or `cli/accept_ballot.py`).

Reply:
I may not be understanding things.  I understand about moving arg parsing back out (up a directory).  However, the src/vtp/ops/foo:main function can't be moved back out to src/vtp/foo:main as that code needs to be executed when one src/vtp/ops/foo1 operation calls another src/vtp/ops/foo2 operations.

I can rename src/vtp/ops/foo:main to src/vtp/ops/foo:fake_main, leave the code in fake_main, and create a thin main function in/as src/vtp/foo:main, but I don't think that is what you are looking for.  Or is it?

?

Reply:
OBE at this point and covered by the conversation 2 above